carbonate_sync.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import errno
+import getpass
+import itertools
+import logging
+import multiprocessing
+import os
+import progressbar
+import shutil
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+import shlex
+import re
+
+from carbon import util
+import carbonate.cli as carbonate_cli
+import carbonate.config as carbonate_config
+import carbonate.cluster as carbonate_cluster
+import carbonate.sieve as carbonate_sieve
+import carbonate.sync as carbonate_sync
+import carbonate.util as carbonate_util
+
+_DEFAULT_TMP_DIR = tempfile.gettempdir()
+STORAGE_DIR = carbonate_cli.STORAGE_DIR
+
+MANDATORY_SSH_OPTIONS = [
+    '-o', 'PasswordAuthentication=no',
+    '-o', 'StrictHostKeyChecking=no',
+]
+
+SSH_OPTIONS = [
+    '-o', 'UserKnownHostsFile=/dev/null',
+    '-o', 'Compression=no',
+    '-o', 'LogLevel=quiet',  # Removes warnings
```

Comment:
This is actually required as warnings bubble up and mess up with the remote commands output (eg: you get a "Warning: Host..." that prevents parsing the list of metrics)